### ISSUE TITLE
Use of clang is hardcoded on macOS and FreeBSD

### ISSUE THREAD

**Issue by ryandesign** [gittup/tup#497]
The use of the `clang` compiler is hardcoded on macOS and FreeBSD:

https://github.com/gittup/tup/blob/8bc40eda2f280f1d5187629bbdb32ad92312e569/build.sh#L50

https://github.com/gittup/tup/blob/8bc40eda2f280f1d5187629bbdb32ad92312e569/build.sh#L56

https://github.com/gittup/tup/blob/8bc40eda2f280f1d5187629bbdb32ad92312e569/freebsd.tup#L1

https://github.com/gittup/tup/blob/8bc40eda2f280f1d5187629bbdb32ad92312e569/macosx.tup#L1

Other operating systems are afforded the courtesy of being able to change the compiler via the `CC` variable:

e4db9199c0dc4452b83088fa2bc7a9503c47f975

On other operating systems the default compiler is `gcc`:

https://github.com/gittup/tup/blob/8bc40eda2f280f1d5187629bbdb32ad92312e569/build.sh#L33

I suggest that on all operating systems the default compiler should be `cc` and that on all operating systems the user should be allowed to override it with `CC`.

**Comment by gittup** [comment]
Good catch, thanks. I've left the default compilers as-is for now, but they are now override-able in the build.sh script for the bootstrap version and via CONFIG_CC in tup.config for the actual tup build on all platforms. (The only exception is for the build-win32.sh script, which is created from 'tup generate').
